src/dom/mod.rs

```diff
@@ -128,7 +128,10 @@ impl Dom {
                     if dom.tree_type == DomVariant::Empty {
                         dom.tree_type = DomVariant::DocumentFragment;
                     }
-                    dom.children.push(Node::Text(pair.as_str().to_string()));
+                    let text = pair.as_str().to_string();
+                    if !text.trim().is_empty() {
```

Comment:
Text nodes now also include only white space characters which we can simply filter out here

Reply:
Tested locally, this breaks html like 
```html
<b>hello</b> <b>world</b>
```